### ISSUE TITLE
FEAT: Add subfolder selection to LoadImage node

### ISSUE THREAD

**Issue by thelemuet** [comfyanonymous/ComfyUI#1220]
Add subfolder selection to LoadImage node, allowing to list images from ComfyUI/input/subfolders (if any).

![imgload](https://github.com/comfyanonymous/ComfyUI/assets/86247933/b4c5cb0d-1494-4f1b-9c87-05d801e5f61f)

For people like me who almost exclusively do img2img the input/ folder gets really messy very quickly, easily can get to multiple hundreds of files in there with no way to categorize.

I could be mistaken but I believe this addition would not really add complexity or confusion to users, simply set <none> as default and it can be ignored.

There are some great custom nodes that I use a lot such as Image Sequence Loader by melMass, that allow to enter custom path, perfect for files with specific naming convention (ie: animation sequence). But for general purpose I have not found a custom node solution, hence the feature suggestion.

**Comment by HiProAI** [comment]
Any update on this anyone?

**Comment by pto2k** [comment]
The default LoadImage node does not support subfolders. It also does not support preview images before selecting them. It becomes very uncomfortable to use when there are a lot of images in the folder, all with non-descriptive file names.

My current solution is to use the ImageLoad node from WAS Node Suit instead. I browse images in the Input folder or other folders using Double Commander because it has a shortcut key to copy the file path. Then I can paste the path into the ImageLoad node quickly.

Are there any other good solutions?

**Comment by HaydenReeve** [comment]
A quick update from me, I am working on this implementation now. It's slow progress, but I am hoping to finish this month along with a few other string quality of life improvements.

**Comment by HiProAI** [comment]
> A quick update from me, I am working on this implementation now. It's slow progress as my bucket is particularly full at the moment, but I am hoping to finish this month along with a few other string/path quality of life improvements.
@HaydenReeve 
Did you ever get around to finishing this? 
Thanks

**Comment by ltdrdata** [comment]
In my opinion, I think it would be better to create this as a separate custom node.

**Comment by HaydenReeve** [comment]
> > A quick update from me, I am working on this implementation now. It's slow progress as my bucket is particularly full at the moment, but I am hoping to finish this month along with a few other string/path quality of life improvements.

> @HaydenReeve
> Did you ever get around to finishing this?
> Thanks

No. I got a rudimentary version working for myself, but I've had my focus torn away from it for a while now. I've ended up using pysss scripts to handle checkpoint/lora loading which has a nice little dropdown feature. Not quite what I was after, but it's doing the job until I can get around to finishing my own experiments.

**Comment by LukeG89** [comment]
[Crystools](https://github.com/crystian/ComfyUI-Crystools) already implemented this feature with the `Load image with metadata`:

![crystools_image_loader](https://github.com/user-attachments/assets/3787a503-087c-464d-b2c9-95e781c8f14b)

And the subfolders like you see is thanks to [rgthree](https://github.com/rgthree/rgthree-comfy) `Auto Nest Subdirectories` feature:
![rgthree_auto_nest](https://github.com/user-attachments/assets/2f47a74a-ebe2-40be-b47c-37c5934cd8dc)

**Comment by freeo** [comment]
While Crystools + Auto Nesting is nice, it doesn't follow symlinks. I'd love for the subfolders to support symlinks, so that all my potential input images don't have to live inside of comfyui/input. This would make management of big image collections way easier.

**Comment by xbourque** [comment]
+1 !!!

It is very frustrating as a user to have to rely on third-party custom nodes just to be allowed to organise images in my 'input' folder into subfolders.

**Is it possible to have the Load Image node behave the same as the Load Checkpoint node** (i.e. show all the files recursively including files in subfolders)?

This doesn't require fancy UI updates or changes to the current behaviour of the Load Image node but would make life MUCH easier for everyone.

**Comment by molbal** [comment]
Just submitted a PR for this change: https://github.com/Comfy-Org/ComfyUI/pull/12099
